api/v1alpha1/inferencemodel_types.go

```diff
@@ -17,71 +17,82 @@ limitations under the License.
-	PoolRef []corev1.ObjectReference `json:"poolRef,omitempty"`
-}
-
-// Model defines the policies for routing the traffic of a use case, this includes performance objectives
-// and traffic splitting between different versions of the model.
-type Model struct {
+type InferenceModelSpec struct {
 	// The name of the model as the users set in the "model" parameter in the requests.
-	// The name should be unique among the services that reference the same backend pool.
+	// The name should be unique among the workloads that reference the same backend pool.
 	// This is the parameter that will be used to match the request with. In the future, we may
 	// allow to match on other request parameters. The other approach to support matching on
 	// on other request parameters is to use a different ModelName per HTTPFilter.
 	// Names can be reserved without implementing an actual model in the pool.
 	// This can be done by specifying a target model and setting the weight to zero,
 	// an error will be returned specifying that no valid target model is found.
-	Name string `json:"name,omitempty"`
+	ModelName string `json:"modelName,omitempty"`
 	// Optional
-	// LLM Services with an objective have higher priority than services without.
-	// IMPORTANT: By specifying an objective, this places the LLMService in a higher priority class than LLMServices without a defined priority class.
-	// In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
-	Objective *Objective `json:"objective,omitempty"`
+	// Defines how important it is to serve the model compared to other models referencing the same pool.
+	Criticality *Criticality `json:"criticality,omitempty"`
 	// Optional.
 	// Allow multiple versions of a model for traffic splitting.
 	// If not specified, the target model name is defaulted to the modelName parameter.
 	// modelName is often in reference to a LoRA adapter.
 	TargetModels []TargetModel `json:"targetModels,omitempty"`
+	// Reference to the InferencePool that the model registers to. It must exist in the same namespace.
+	PoolRef LocalObjectReference `json:"poolRef,omitempty"`
 }
 
+// LocalObjectReference identifies an API object within the namespace of the
+// referrer.
+type LocalObjectReference struct {
+	// Group is the group of the referent.
+	Group string `json:"group,omitempty"`
+
+	// Kind is kind of the referent. For example "InferencePool".
+	Kind string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	Name string `json:"name,omitempty"`
+}
+
+// Defines how important it is to serve the model compared to other models.
+type Criticality string
+
+const (
+	// Most important. Requests to this band will be shed last.
+	Critical Criticality = "Critical"
+	// More important than Sheddable, less important than Critical.
+	// Requests in this band will be shed before critical traffic.
+	Default Criticality = "Default"
```

Comment:
I don't hate this name, but it always feels a bit funny to have to spell out "Default". Would something like "Moderate" or "Normal" work better here? Maybe there are some similar terms we can use as a reference? For example Kubernetes Pods have the following QoS classes:

* Guaranteed
* Burstable
* BestEffort

Reply:
@JeremyOT if you have thoughts on what other names we can use for "Default". Another suggestion is "Standard"? I can also get behind "Moderate" or "Normal".

Reply:
`Moderate` sounds good to me here, seems at least better than `Default`

Reply:
I actually like Default, because we are really trying to communicate that this is what you get when unset. No docs necessary to understand that. Something like Moderate is fine, but I still need to remember that Moderate is the default. Normal is more clear.

I don't think K8s QoS is a great comparison, since there's so much difference between behaviors of each class. It's not just more/less.

Reply:
What about representing these in numbers like priorities instead of abstract names? Everything is relative to each other.

Reply:
I think defining a discrete value is sufficient, easier for users to work with (pre-defined options to select from) and makes provider implementations easier.

Reply:
I can get on board with starting with 3 simple values here, but likely worth revisiting in the future to see if additional granularity would be helpful. @terrytangyuan did you have any specific use cases in mind that would benefit from having additional levels here?